src/commands/synthetics/mobile.ts

```diff
@@ -83,24 +83,28 @@ export const uploadApplicationAndOverrideConfig = async (
   overriddenTestsToTrigger: TestPayload,
   uploadedApplicationByPath: {[applicationFilePath: string]: {applicationId: string; fileName: string}[]}
 ): Promise<void> => {
-  if (!userConfigOverride.mobileApplicationVersionFilePath) {
+  if (!(userConfigOverride.mobileApplicationVersionFilePath || userConfigOverride.mobileApplicationVersion)) {
     return
   }
 
   const testApplicationId = test.options.mobileApplication!.applicationId
-  await uploadApplicationIfNeeded(
-    api,
-    userConfigOverride.mobileApplicationVersionFilePath,
-    testApplicationId,
-    uploadedApplicationByPath
-  )
+  if (userConfigOverride.mobileApplicationVersionFilePath) {
```

Comment:
Since the method called below is `uploadApplicationIfNeeded()`, I would say that the check if should upload or not should be done inside it! Can you move this check inside the method?

Maybe make [uploadedApplicationByPath](https://github.com/DataDog/datadog-ci/blob/5a36bce2391a544367e1b22a11834ce66680475c/src/commands/synthetics/mobile.ts#L34) as optional, and check if it's not set.

Reply:
IMO, this check shouldn't be done in `uploadApplicationIfNeeded`. `uploadApplicationIfNeeded` uploads the application if it is not cached. We can enhance the naming with `uploadApplicationIfNotCached` ?

Reply:
Humm IMO `ifNeeded` gives the idea of checking the edge cases, and only doing the upload "if needed".

IMHO we can just say then `uploadApplication()`, and skip if already cached. No need to keep it in the method name!

Another option which would be great for unit testing is having a `shouldUploadApplication()` which checks for all these constraints (if there is an override, if the file is already cached, etc), and let `uploadApplication()` only perform the upload 👍